src/NHibernate/Mapping/Constraint.cs

```diff
@@ -101,10 +101,10 @@ public virtual string SqlDropString(Dialect.Dialect dialect, string defaultCatal
 		{
 			if (IsGenerated(dialect))
 			{
-				string ifExists = dialect.GetIfExistsDropConstraint(Table, Name);
-				string drop =
-					string.Format("alter table {0} drop constraint {1}", Table.GetQualifiedName(dialect, defaultCatalog, defaultSchema), Name);
-				string end = dialect.GetIfExistsDropConstraintEnd(Table, Name);
+				var tableName = Table.GetQualifiedName(dialect, defaultCatalog, defaultSchema);
+				string ifExists = dialect.GetIfExistsDropConstraint(Table, Name, defaultCatalog, defaultSchema);
+				string drop = string.Format("alter table {0} drop constraint {1}", Table.GetQualifiedName(dialect, defaultCatalog, defaultSchema), Name);
+				string end = dialect.GetIfExistsDropConstraintEnd(Table, tableName, defaultCatalog, defaultSchema);
 
```

Comment:
It could be part of a refactoring. I think we should just close this one with the current solution.